### PR TITLE
Implement count() method for Iterator

### PR DESCRIPTION
**Pull request by belochub**

### CODE REVIEW

File: lib/iterator.js [hunk 1]
```diff
@@ -22,6 +22,14 @@ class Iterator {
     return this.base.next();
   }
 
+  size() {
```
Comment (nechaido) [inline lib/iterator.js]: I'd prefer to name it `amount` or `count`.

Comment (nechaido) [reply lib/iterator.js]: Maybe even `leftUnconsumed`.

File: test/iterator.js [hunk 2]
```diff
@@ -29,6 +29,21 @@ metatests.test('Iterator is Iterable', test => {
   test.end();
 });
 
+metatests.test('Iterator.size', test => {
+  const iterator = iter(array);
+
+  test.strictSame(iterator.size(), array.length);
```
Comment (nechaido) [inline test/iterator.js]: I'd prefer: `iter(array).size()`.

File: test/iterator.js [hunk 3]
```diff
@@ -29,6 +29,21 @@ metatests.test('Iterator is Iterable', test => {
...
+  test.end();
+});
+
+metatests.test('Iterator.size on consumed iterator', test => {
+  const iterator = iter(array);
+  iterator.skip(array.length);
+
+  test.strictSame(iterator.size(), 0);
```
Comment (nechaido) [inline test/iterator.js]: I'd prefer: `iter(array).skip(array.length).size()`.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/common/commit/1c683b2e1723e1c3a64563f64b4a51aa2853da45.

**Pull request closed without merging**